### PR TITLE
feat(vertexai): Elastic Text-Embedding Model demo.

### PR DESCRIPTION
**Pull request by kewent**
## Description

Fixes: http://b/334964454

## Checklist
- [x] I have followed [Contributing Guidelines from CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md)
- [x] **Tests** pass:   `go test -v ./..` (see [Testing](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#testing))
- [x] **Code formatted**:   `gofmt` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [x] **Vetting** pass:   `go vet` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [x] Please **merge** this PR for me once it is approved

### CODE REVIEW

**Review by gericdong — APPROVED** [review]

**Comment by grayside** [comment]
This PR is modifying a sample that's currently in conflict with https://github.com/GoogleCloudPlatform/golang-samples/blob/main/aiplatform/text-embeddings/embeddings.go. One of these samples should be removed or the region tag changed.

File: aiplatform/snippets/embeddings.go [hunk 1]
```diff
@@ -28,7 +28,8 @@ import (
 )
 
 func embedTexts(
-	apiEndpoint, project, model string, texts []string, task string) ([][]float32, error) {
+	apiEndpoint, project, model string, texts []string,
+	task string, customOutputDimensionality *int) ([][]float32, error) {
```
Comment (grayside) [inline aiplatform/snippets/embeddings.go:32]: suggestion: These parameters need additional usage documentation and several should possibly be bundled into a struct. See #4110 and #4113 for examples.

Comment (grayside) [inline aiplatform/snippets/embeddings.go:32]: From https://googlecloudplatform.github.io/samples-style-guide/#minimal-arguments, we should be using minimal arguments.

blocker: `apiEndpoint` is generally not specified in samples and should be removed. Looking at the code, it seems like maybe the problem is there's a missing method in the library to facilitate text embedding? If so we can make this not-blocker.

File: aiplatform/snippets/embeddings_test.go [hunk 2]
```diff
@@ -23,14 +23,15 @@ import (
...
 	texts := []string{"banana muffins? ", "banana bread? banana muffins?"}
-	embeddings, err := embedTexts(apiEndpoint, tc.ProjectID, model, texts, "RETRIEVAL_DOCUMENT")
+	dimensionality := 5
+	embeddings, err := embedTextsPreview(apiEndpoint, tc.ProjectID, model, texts, "QUESTION_ANSWERING", &dimensionality)
 	if err != nil {
 		t.Fatal(err)
 	}
-	if len(embeddings) != len(texts) || len(embeddings[0]) != 768 {
-		t.Errorf("len(embeddings), len(embeddings[0]) = %d, %d, want %d, 768", len(embeddings), len(embeddings[0]), len(texts))
+	if len(embeddings) != len(texts) || len(embeddings[0]) != dimensionality {
```
Comment (grayside) [inline aiplatform/snippets/embeddings_test.go:33]: suggestion: consider improving readability by extracting repeating `len` calls to 'got' variables.

File: aiplatform/snippets/embeddings.go (hunk 1 above)
Comment (grayside) [inline aiplatform/snippets/embeddings.go:32]: From https://googlecloudplatform.github.io/samples-style-guide/#minimal-arguments, we should be using minimal arguments.

suggestion: Remove `customOutputDimensionality` as a parameter in favor of hard-coding in the function body.

**Review by grayside — CHANGES_REQUESTED** [review]

File: aiplatform/snippets/embeddings.go (hunk 1 above)
Comment (kewent) [reply aiplatform/snippets/embeddings.go:32]: SG, I'll hard-coded some of them and only leave `project` and `texts.`

Comment (kewent) [reply aiplatform/snippets/embeddings.go:32]: SG

Comment (kewent) [reply aiplatform/snippets/embeddings.go:32]: SG

**Comment by kewent** [comment]
> This PR is modifying a sample that's currently in conflict with https://github.com/GoogleCloudPlatform/golang-samples/blob/main/aiplatform/text-embeddings/embeddings.go. One of these samples should be removed or the region tag changed.

Removed duplicated files under ./text-embeddings/

**Review by grayside — APPROVED** [review]

**Comment by grayside** [comment]
I just checked https://pkg.go.dev/cloud.google.com/go/aiplatform/apiv1beta1/aiplatformpb#DirectPredictRequest, and realized specifying the Endpoint is required. This is not typical and I assumed it could be replaced with a direct "Location" setting. I'm sure that meant my previous feedback about Endpoint/Location was confusing. Thank you for your patience through my misunderstanding of the API design.

**Pull request merged**